policy/policy.go

```diff
@@ -97,21 +102,69 @@ func (l UpdatedList) Get(domain string) (TLSPolicy, error) {
 	return l.get(domain)
 }
 
+// Raw returns a raw List struct, copied from the underlying one
+func (l UpdatedList) Raw() List {
+	l.mu.RLock()
+	defer l.mu.RUnlock()
+	list := List{
+		Timestamp:     l.Timestamp,
+		Expires:       l.Expires,
+		Version:       l.Version,
+		Author:        l.Author,
+		Pinsets:       make(map[string]Pinset),
+		PolicyAliases: make(map[string]TLSPolicy),
+		Policies:      make(map[string]TLSPolicy),
+	}
+	for pinName, pinset := range l.Pinsets {
+		list.Pinsets[pinName] = pinset.clone()
+	}
+	for alias, policy := range l.PolicyAliases {
+		list.PolicyAliases[alias] = policy.clone()
+	}
+	for domain, policy := range l.Policies {
+		list.Policies[domain] = policy.clone()
+	}
+	return list
+}
+
+func (p Pinset) clone() Pinset {
+	pinset := Pinset{StaticSPKIHashes: make([]string, 0)}
+	for _, hash := range p.StaticSPKIHashes {
+		pinset.StaticSPKIHashes = append(pinset.StaticSPKIHashes, hash)
+	}
+	return pinset
+}
+
+func (p TLSPolicy) clone() TLSPolicy {
```

Comment:
Wow, making a deep copy in Go is not very straightfoward! I agree we need to manually clone the nested structs/slices.

By my reading of this [blog post](https://flaviocopes.com/go-copying-structs/#deep-copy-a-struct), it seems that we could pare this down a bit by taking advantage of the fact that Go copies basic types by value when a struct is assigned to a new variable. That would apply to basic types in `list` as well.